### DISCUSSION TITLE
Allow moving combo counter to mania stage

### DISCUSSION THREAD

**Discussion by 04meta** [ppy/osu#13017] [Ideas]
**Describe the feature:** Even if we don't add it to any existing gamemodes (maybe osu!mania?), it would be nice to have it in place so anyone making a custom gamemode doesn't have to manually edit dozens of source files to add it (perhaps to make their skin look more like [IIDX](https://www.youtube.com/watch?v=yW34iG4bsQw)).

**Proposal designs of the feature:** Well, it would be nice if osu! didn't break the PrintScreen key (is this some sort of anti-cheating mechanism?) Anyway, the combo counter to the right of my score is apparently a rolling counter that hadn't updated yet.

![An osu!mania Normal difficulty with HalfTime on, with the latest judgment being "Perfect 11" and the combo counter about to update to 11.](https://user-images.githubusercontent.com/38385509/53591062-c92f6c80-3b58-11e9-8887-83a2d2513e23.jpg)

**Comment by peppy** [comment]
I can't understand what you are asking for here. Also osu! does not do anything to print screen functionality, but you can hit f12 to take an in-game screenshot.

**Comment by smoogipoo** [comment]
The OP wants to display the combo number besides the judement as in the IIDX video mentioned.

After discussion, we'll probably want to handle this in a different way than giving `JudgementResult` a `ComboAfterJudgement` or similar, and instead create a combo counter that is reusable for all scenarios and can be pick-and-placed wherever desired through skinning.

Likely a good starting point is to cache the `ScoreProcessor` and DI it into a `ComboCounter` class, and have it bind to the combo bindable internally.
